Stubbing

thatnoobles opened the pull request:
- Stubbed out most of the lower-level I/O classes.
- Added docstrings for things where we have a good idea of what they're going to do. A few have placeholders for now

cvr8924 commented:
Make cpp file

thatnoobles committed:
stubbed out motor & pneumatics classes

thatnoobles committed:
stubbed out some of receiver

thatnoobles committed:
created basic adruino header files

thatnoobles committed:
stubbed out states for possible eventhandler + log

thatnoobles committed:
added basic docstrings for (most) things

hiromon0125 commented on src/arduino/comms/packager.h:
```diff
@@ -0,0 +1,20 @@
+#include <pressure.h>
+#include <potentiometer.h>
+#include <packet.h>
+
```
```suggestion
// TODO: docstring
```

hiromon0125 commented on src/raspi/io/comms/receiver.py:
```diff
@@ -0,0 +1,6 @@
+from potentiometer.pot import Potentiometer
+from pneumatics.pressure import Pressure
+
+current_data : list[int]
+potentiometers : list[Potentiometer]
+pressure : Pressure
```
use SCREAM_SNAKE_CASE

hiromon0125 commented on src/raspi/io/pneumatics/pressure.py:
```diff
@@ -0,0 +1,26 @@
+    Represents air pressure readings to measure current flexion/extension
+    (knee) movement of each leg.
+    """
+    MIN_PRESSURE : float = 0.0  # TODO: assign this
+    MAX_PRESSURE : float = 0.0  # and this
```
should we put this outside of pressure class instead?

hiromon0125 commented on src/raspi/io/pneumatics/valve.py:
```diff
@@ -0,0 +1,30 @@
+        """
+        ...
+
+    def set_valve(state : bool) -> None:
```
```suggestion
    def set_valve(self, state : bool) -> None:
```

hiromon0125 commented:
couple of docstring and variable changes but overall lgtm

thatnoobles committed:
Update src/raspi/io/pneumatics/valve.py

set_value should pass 'self' as a parameter

Co-authored-by: Hiroto Takeuchi <hiroto125takeuchi@gmail.com>

thatnoobles committed:
Update src/arduino/comms/packager.h

Little TODO message for adding docstrings to the C++ code if needed eventually

Co-authored-by: Hiroto Takeuchi <hiroto125takeuchi@gmail.com>
